src/new-components/notifications/notifications-animation.ts

```diff
@@ -0,0 +1,187 @@
+/**
```

Comment:
I was expecting maybe just something [like](https://github.com/reactjs/react-transition-group) this to add a class-based enter/exit transition - not sure if it's totally possible to implement the stacking exit transition you have now that way though. 

I'd feel better if this came in the form of a library, as by being in the codebase this becomes something we have to maintain, and possibly diverge further from the original implementation - but the animation looks good, so let's go with it for now.